services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/resource_tracker_core.py

```diff
@@ -46,11 +52,38 @@ def _prometheus_sync_client_custom_query(
     return data
 
 
+def _build_cache_key_user_id(fct, *args):
+    return f"{fct.__name__}_{args[1]}"
+
+
+def _build_cache_key_project_and_node_id(fct, *args):
+    return f"{fct.__name__}_{args[1]}_{args[2]}"
+
+
+@cached(ttl=_TTL, key_builder=_build_cache_key_user_id)
+async def _get_user_email(
+    osparc_repo: UserAndProjectRepository, user_id: int
+) -> str | None:
+    user_email: str | None = await osparc_repo.get_user_email(user_id)
+    return user_email
+
+
+@cached(ttl=_TTL, key_builder=_build_cache_key_project_and_node_id)
+async def _get_project_and_node_names(
+    osparc_repo: UserAndProjectRepository, project_uuid: ProjectID, node_uuid: NodeID
+) -> tuple[str | None, str | None]:
+    if output := await osparc_repo.get_project_name_and_workbench(project_uuid):
+        project_name, project_workbench = output
+        return (project_name, project_workbench[f"{node_uuid}"].get("label"))
+    return (None, None)
```

Comment:
Question: one thing is not clear to me. Is there an issue if these values get out of date with the read ones?

Reply:
For now, it should not be an issue, this is just additional data that is stored there (basically data that you started the service with). If in the future for some reason this logic needs to be changed there are still options for how to update the data.